cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/RunsCount/index.tsx

```diff
@@ -15,27 +15,31 @@
  */
 
 import * as React from 'react';
-import { connect } from 'react-redux';
-import { IRunsCountMap, IPipeline } from 'components/PipelineList/DeployedPipelineView/types';
+import { IApplicationRecord } from 'components/PipelineList/DeployedPipelineView/types';
+import { getProgram } from 'components/PipelineList/DeployedPipelineView/graphqlHelper';
+import { objectQuery } from 'services/helpers';
 
 interface IProps {
-  runsCountMap: IRunsCountMap;
-  pipeline: IPipeline;
+  pipeline: IApplicationRecord;
 }
 
-const RunsCountView: React.SFC<IProps> = ({ runsCountMap, pipeline }) => {
-  const runsCount = runsCountMap[pipeline.name] || 0;
+const RunsCountView: React.SFC<IProps> = ({ pipeline }) => {
+  const runsCount = getRunsCount(pipeline);
 
   return <div className="runs">{runsCount}</div>;
 };
 
-const mapStateToProps = (state, ownProp) => {
-  return {
-    runsCountMap: state.deployed.runsCountMap,
-    pipeline: ownProp.pipeline,
-  };
-};
+function getRunsCount(pipeline) {
+  const program = getProgram(pipeline);
+  const totalRuns = objectQuery(program, 'totalRuns');
+
+  if (totalRuns === null || totalRuns === undefined) {
+    return 0;
+  }
```

Comment:
you can simply do `return totalRuns || 0;`

Reply:
Done